src/Microsoft.AspNetCore.CertificateGeneration.Task/CertificateManager.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+
+namespace Microsoft.AspNetCore.CertificateGeneration.Task
+{
+    internal static class CertificateManager
+    {
+        public static X509Certificate2 GenerateSSLCertificate(
+            string subjectName,
+            IEnumerable<string> subjectAlternativeName,
+            string friendlyName,
+            DateTimeOffset notBefore,
+            DateTimeOffset expires,
+            StoreName storeName,
+            StoreLocation storeLocation)
+        {
+            using (var rsa = RSA.Create(2048))
+            {
+                var signingRequest = new CertificateRequest(
+                    new X500DistinguishedName(subjectName), rsa, HashAlgorithmName.SHA256, RSASignaturePadding.Pkcs1);
+
+                var enhancedKeyUsage = new OidCollection();
+                enhancedKeyUsage.Add(new Oid("1.3.6.1.5.5.7.3.1", "Server Authentication"));
+                signingRequest.CertificateExtensions.Add(new X509EnhancedKeyUsageExtension(enhancedKeyUsage, critical: true));
+                signingRequest.CertificateExtensions.Add(new X509KeyUsageExtension(X509KeyUsageFlags.KeyEncipherment, critical: true));
+                signingRequest.CertificateExtensions.Add(
+                    new X509BasicConstraintsExtension(
+                        certificateAuthority: false,
+                        hasPathLengthConstraint: false,
+                        pathLengthConstraint: 0,
+                        critical: true));
+
+                var sanBuilder = new SubjectAlternativeNameBuilder();
+                foreach (var alternativeName in subjectAlternativeName)
```

Comment:
@blowdart as this PR has been merged already, should we open a new issue to fix this before preview2?